code/modules/power/battery.dm

```diff
@@ -192,12 +197,15 @@ var/global/list/battery_online =	list(
 	data["storedCapacity"] = round(100.0*charge/capacity, 0.1)
 	data["charging"] = charging
 	data["chargeMode"] = chargemode
+	data["chargeLoad"] = round(loadcharge)
 	data["chargeLevel"] = chargelevel
-	data["chargeMax"] = smes_input_max
+	data["chargeMax"] = max_input
 	data["outputOnline"] = online
 	data["outputLevel"] = output
-	data["outputMax"] = smes_output_max
+	data["outputMax"] = max_output
 	data["outputLoad"] = round(loaddemand)
+	data["hasInput"] = terminal ? 1 : 0;
```

Comment:
This won't work for portable SMESs, the terminal is owned by a different machine

Reply:
A bit busy IRL, will look into in a couple of days.